core/src/main/java/io/questdb/griffin/engine/table/LatestByAllFilteredRecordCursorFactory.java

```diff
@@ -52,6 +54,7 @@ public LatestByAllFilteredRecordCursorFactory(
     ) {
         super(metadata, dataFrameCursorFactory, configuration);
         try {
+            this.filter = filter;
```

Comment:
this was the bugfix. Filter must never be closed by the cursor, otherwise cursor is not reusable